lbrynet/wallet/database.py

```diff
@@ -63,7 +63,10 @@ def txo_to_row(self, tx, address, txo):
                 if txo.claim.is_signed:
                     channel_ids.add(txo.claim.signing_channel_id)
                 if txo.claim.is_channel and my_account is not None:
-                    txo.private_key = my_account.get_channel_private_key(txo.ref)
+                    channel_pubkey_hash = my_account.ledger.public_key_to_address(
+                        txo.claim.channel.public_key_bytes
+                    )
+                    txo.private_key = my_account.get_channel_private_key(channel_pubkey_hash)
```

Comment:
can `get_channel_private_key` take a txo and do the `public_key_to_address()` conversion internally?

Reply:
Yes, but since not all txos are channel txos this will have to have error handling and all that which is already handled in database.py in the sense that it only sends the correct value in the first place. I think this is better/fine.